src/org/aavso/tools/vstar/ui/dialog/plugin/manager/PluginManager.java

```diff
@@ -881,30 +891,43 @@ private void copy(InputStream in, File file) throws IOException {
 		}
 	}
 
-	// return class loader here!
-	private Pair<IPlugin, URLClassLoader> createObjectFromJarURL(URL url, String className)
-			throws ClassNotFoundException, IllegalAccessException,
-			InstantiationException {
-		URLClassLoader loader = new URLClassLoader(new URL[] { url },
-				VStar.class.getClassLoader());
-		Class<?> clazz = loader.loadClass(className);
-		IPlugin plugin = (IPlugin) clazz.newInstance(); 
-		return new Pair<IPlugin, URLClassLoader>(plugin, loader); 
+	private String getPluginDescription(URL url, String className, List<URL> depLibs)
+		throws IOException, ClassNotFoundException, InstantiationException, IllegalAccessException {
+		List<URL> urlList = new ArrayList<URL>();
+		urlList.add(url);
+		urlList.addAll(depLibs);
+		URL[] urls = urlList.toArray(new URL[0]);
+		URLClassLoader loader = new URLClassLoader(urls, VStar.class.getClassLoader());		
+		try {
+			Class<?> clazz = loader.loadClass(className);
+			IPlugin plugin = (IPlugin) clazz.newInstance();
+			return plugin.getDescription();
+		} finally {
+			loader.close();
+		}
 	}
 
 	private boolean areURLReferentsEqual(URL url1, URL url2) throws IOException {
-		return getBytesFromURL(url1).equals(getBytesFromURL(url2));
+		// Max: arrays retrieved by getBytesFromURL will never be equal!
```

Comment:
True! Good pick up. There should really be no `equals()` method for arrays in Java since this and `==` are identical.